app/com/arpnetworking/metrics/portal/reports/ReportRepository.java

```diff
@@ -45,5 +49,23 @@
      * @param organization The {@code Organization} which owns the report.
      */
     void addOrUpdateReport(Report report, Organization organization);
+
+    /**
+     * Query reports.
+     *
+     * This is a specialization of {@link JobRepository#query} for cases where you want to
+     * handle the reports themselves, outside of a Job context.
+     *
+     * @param query The {@code JobQuery} instance to execute.
+     * @return The reports resulting from executing the query.
+     */
+    QueryResult<Report> queryReports(JobQuery<Report.Result> query);
```

Comment:
Should this accept a `ReportQuery`? Initially, this could just be an empty subclass of `JobQuery`, but it would allow for report specific query fields to be specified when dealing with reports queries. For example, you could query by title of report (which does not apply to job). Of course this does mean that you would have to be able to create a `ReportQuery` from a `JobQuery` in `query` below. I don't think it's necessary for Spencer's usage, but it depends on what you had in mind for the list user interface.

Reply:
I thought this would work but now I'm thinking it shouldn't be a subclass. I agree that we should have a separate type to allow for filtering by Report-specific fields, but as a subclass the `JobQuery#execute` method becomes less useful (if I used a `ReportQuery#execute` it would still return a Job which isn't what I want)